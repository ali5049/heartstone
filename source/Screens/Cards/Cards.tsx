import React from 'react';
import { FlatList, SafeAreaView, Text } from 'react-native';
import { ICard } from '../../@types/card.types';
import { CardsProps } from '../../@types/navigation.types';
import { Card } from '../../Components';

const ListEmptyComponent: React.FC = () => <Text>No cards available</Text>;

const Cards: React.FC<CardsProps> = ({route}) => {
  const cards: ICard[] = route.params.cards || [];

  return (
    <SafeAreaView>
      <FlatList
        ListEmptyComponent={ListEmptyComponent}
        data={cards}
        renderItem={({item}) => <Card card={item} />}
      />
    </SafeAreaView>
  );
};

export default Cards;
