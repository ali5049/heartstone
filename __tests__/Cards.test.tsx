import React from 'react';
import {render} from '@testing-library/react-native';
import Cards from '../source/Screens/Cards';
import {ICard} from '../source/@types/card.types';

describe('Cards Component', () => {
  const mockCards: ICard[] = [
    {
      cardId: '342342',
      dbfId: 3432,
      name: 'Inner Range',
      cardSet: 'Basic',
      type: 'Spell',
      faction: 'Neutral',
      rarity: 'Free',
      cost: 5,
      playerClass: 'Paladin',
      locale: 'enUS',
    },
  ];
  const mockRoute = {
    params: {cards: mockCards},
    key: '1',
    name: 'Cards' as 'Cards',
  };
  const mockRoute2 = {
    params: {cards: []},
    key: '2',
    name: 'Cards' as 'Cards',
  };

  it('renders a list of cards', () => {
    const {getByText} = render(<Cards route={mockRoute} />);
    expect(getByText('Name: Inner Range')).toBeTruthy();
  });

  it('shows an empty state when there are no cards', () => {
    const {getByText} = render(<Cards route={mockRoute2} />);
    expect(getByText('No cards available')).toBeTruthy();
  });

  test('Cards Component renders correctly', () => {
    const {toJSON} = render(<Cards route={mockRoute2} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
