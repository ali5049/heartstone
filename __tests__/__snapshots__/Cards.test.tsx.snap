// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Component Cards Component renders correctly 1`] = `
<RCTSafeAreaView>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <Text>
        No cards available
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
